presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -780,4 +747,89 @@ private PlanNode distinct(PlanNode node)
                 1.0,
                 Optional.empty());
     }
+
+    private class SetOperationPlanner
```

Comment:
leaving it as-is, since this uses some relationPlanner methods. not sure if we should make then static.